pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -73,6 +76,44 @@ func (n *nsLifecycle) syncNS(obj *v1.Namespace) (bool, error) {
 	return hasPRTBs, nil
 }
 
+func (n *nsLifecycle) assignToSystemProject(ns *v1.Namespace) error {
+	defaultProjectsToNamespaces, err := GetDefaultProjectsToNamespaces()
+	if err != nil {
+		return err
+	}
+	for projectDisplayName, namespaces := range defaultProjectsToNamespaces {
+		for _, nsToCheck := range namespaces {
+			if nsToCheck == ns.Name {
+				projectID := ns.Annotations[projectIDAnnotation]
+				if projectID != "" {
+					return nil
+				}
+
+				projects, err := n.m.projectLister.List(n.m.clusterName, labels.NewSelector())
+				if err != nil {
+					return err
+				}
+				var project *v3.Project
+				for _, p := range projects {
+					if p.Spec.DisplayName == projectDisplayName {
```

Comment:
As mentioned previously, keying off of display name for decision making purposes is not a good idea. I think this should be done based off of an annotation or label

Reply:
Updated everywhere to fetch projects based on label `"authz.management.cattle.io/system-project": "true"`